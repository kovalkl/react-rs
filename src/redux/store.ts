import { configureStore } from '@reduxjs/toolkit';

import { peopleApi } from './peopleApi';

export const store = configureStore({
  reducer: {
    [peopleApi.reducerPath]: peopleApi.reducer,
  },
  middleware: (getDefaultMiddleware) => getDefaultMiddleware().concat(peopleApi.middleware),
});
