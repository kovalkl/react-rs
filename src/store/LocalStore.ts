const STORE_KEYS = {
  SEARCH_HISTORY: 'search_history-kovalkl',
};

class LocalStore {
  private static instance: LocalStore;

  public static getInstance(): LocalStore {
    if (!LocalStore.instance) {
      LocalStore.instance = new LocalStore();
    }

    return LocalStore.instance;
  }

  public setSearchHistory = (searchHistory: string): void => {
    this.setStore(STORE_KEYS.SEARCH_HISTORY, searchHistory);
  };

  public getSearchHistory = (): string => {
    return this.getStore(STORE_KEYS.SEARCH_HISTORY);
  };

  public hasSearchHistory = (): boolean => {
    return this.hasStore(STORE_KEYS.SEARCH_HISTORY);
  };

  public removeSearchHistory = (): void => {
    this.removeStore(STORE_KEYS.SEARCH_HISTORY);
  };

  private getStore = (key: string): string => {
    return localStorage.getItem(key)!;
  };

  private hasStore = (key: string): boolean => {
    return Boolean(localStorage.getItem(key));
  };

  private setStore = (key: string, value: string): void => {
    localStorage.setItem(key, value);
  };

  private removeStore = (key: string): void => {
    localStorage.removeItem(key);
  };
}

export default LocalStore;
