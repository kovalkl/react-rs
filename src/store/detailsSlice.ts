import { createSlice, PayloadAction } from '@reduxjs/toolkit';

import { Person } from '@/models/types';

type PersonState = {
  person: Person | null;
};

const initialState: PersonState = {
  person: null,
};

const detailsSlice = createSlice({
  name: 'details',
  initialState,
  reducers: {
    addDetails(state, action: PayloadAction<Person>) {
      state.person = action.payload;
    },
    clearDetails(state) {
      state.person = null;
    },
  },
});

export const { addDetails, clearDetails } = detailsSlice.actions;

export default detailsSlice.reducer;
