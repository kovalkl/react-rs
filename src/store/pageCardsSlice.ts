import { createSlice, PayloadAction } from '@reduxjs/toolkit';

import { Person } from '../models/types';

type PageCardsState = {
  list: Person[];
};

const initialState: PageCardsState = {
  list: [],
};

const pageCardsSlice = createSlice({
  name: 'cardsOnPage',
  initialState,
  reducers: {
    addCards(state, action: PayloadAction<Person[]>) {
      state.list = action.payload;
    },
  },
});

export const { addCards } = pageCardsSlice.actions;

export default pageCardsSlice.reducer;
