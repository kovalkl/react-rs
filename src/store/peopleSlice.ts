import { createSlice, PayloadAction } from '@reduxjs/toolkit';

import { Person } from '../models/types';

type PeopleState = {
  list: Person[];
};

const initialState: PeopleState = {
  list: [],
};

const peopleSlice = createSlice({
  name: 'people',
  initialState,
  reducers: {
    togglePerson(state, action: PayloadAction<Person>) {
      state.list.find((person) => person.url === action.payload.url)
        ? (state.list = state.list.filter((person) => person.url !== action.payload.url))
        : state.list.push(action.payload);
    },
    clearState(state) {
      state.list = [];
    },
  },
});

export const { togglePerson, clearState } = peopleSlice.actions;

export default peopleSlice.reducer;
