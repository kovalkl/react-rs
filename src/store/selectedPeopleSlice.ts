import { createSlice, PayloadAction } from '@reduxjs/toolkit';

import { Person } from '../models/types';

type SelectedPeopleState = {
  list: { [url: string]: Person };
};

const initialState: SelectedPeopleState = {
  list: {},
};

const selectedPeopleSlice = createSlice({
  name: 'selectedPeople',
  initialState,
  reducers: {
    togglePerson(state, action: PayloadAction<Person>) {
      Object.keys(state.list).includes(action.payload.url)
        ? delete state.list[action.payload.url]
        : (state.list[action.payload.url] = action.payload);
    },

    clearState(state) {
      state.list = {};
    },
  },
});

export const { togglePerson, clearState } = selectedPeopleSlice.actions;

export default selectedPeopleSlice.reducer;
