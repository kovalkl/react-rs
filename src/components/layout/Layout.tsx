import { useContext } from 'react';

import { ThemeContext } from '../../store/ThemeContext';
import Flyout from '../flyout/Flyout';
import Header from '../header/Header';
import Main from './../main/Main';

const Layout = () => {
  const { isDarkTheme } = useContext(ThemeContext);

  return (
    <div className={`app ${isDarkTheme ? 'dark-theme' : 'light-theme'}`}>
      <Header />
      <Main />
      <Flyout />
    </div>
  );
};

export default Layout;
