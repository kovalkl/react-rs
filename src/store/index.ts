import { configureStore } from '@reduxjs/toolkit';

import { peopleApi } from '@/redux/peopleApi';
import detailsReducer from '@/store/detailsSlice';
import pageCardsReducer from '@/store/pageCardsSlice';
import selectedPeopleReducer from '@/store/selectedPeopleSlice';

const store = configureStore({
  reducer: {
    selectedPeople: selectedPeopleReducer,
    pageCards: pageCardsReducer,
    details: detailsReducer,
    [peopleApi.reducerPath]: peopleApi.reducer,
  },
  middleware: (getDefaultMiddleware) => getDefaultMiddleware().concat(peopleApi.middleware),
});

export default store;

export type RootState = ReturnType<typeof store.getState>;
export type AppDispatch = typeof store.dispatch;
