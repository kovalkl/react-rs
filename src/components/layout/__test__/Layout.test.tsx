import '@testing-library/jest-dom';
import { render } from '@testing-library/react';

import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';

import store from './../../../store/index';
import Layout from './../Layout';

describe('Flyout', () => {
  it('should render correctly', () => {
    const component = render(
      <Provider store={store}>
        <MemoryRouter>
          <Layout />
        </MemoryRouter>
      </Provider>,
    );

    expect(component).toMatchSnapshot();
  });
});
