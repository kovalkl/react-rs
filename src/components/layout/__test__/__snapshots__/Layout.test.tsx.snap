// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Flyout > should render correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="app light-theme"
      >
        <header
          class="header light-theme"
        >
          <div
            class="header__wrapper container"
          >
            <div
              class="search"
            >
              <form
                class="search__form"
              >
                <input
                  class="text-input"
                  placeholder="Search..."
                  type="text"
                  value=""
                />
                <button
                  class="button  undefined"
                  type="button"
                >
                  Search
                </button>
              </form>
            </div>
            <form
              class="theme-button"
            >
              <label
                class="check-label"
              >
                <input
                  checked=""
                  name="theme"
                  type="radio"
                  value="light"
                />
                Light
              </label>
              <label
                class="check-label"
              >
                <input
                  name="theme"
                  type="radio"
                  value="dark"
                />
                Dark
              </label>
            </form>
          </div>
        </header>
        <main
          class="main"
        >
          <div
            class="main__wrapper container"
          >
            <section
              class="content"
            >
              <h2>
                Loading...
              </h2>
            </section>
          </div>
        </main>
        <div
          class="flyout light-theme "
        >
          <div
            class="container flyout__wrapper"
          >
            <p
              class="flyout__text"
            >
              0 people are selected
            </p>
            <button
              class="button  flyout__btn"
              type="button"
            >
              Download
            </button>
            <button
              class="button button_error flyout__btn"
              type="button"
            >
              Unselect all
            </button>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="app light-theme"
    >
      <header
        class="header light-theme"
      >
        <div
          class="header__wrapper container"
        >
          <div
            class="search"
          >
            <form
              class="search__form"
            >
              <input
                class="text-input"
                placeholder="Search..."
                type="text"
                value=""
              />
              <button
                class="button  undefined"
                type="button"
              >
                Search
              </button>
            </form>
          </div>
          <form
            class="theme-button"
          >
            <label
              class="check-label"
            >
              <input
                checked=""
                name="theme"
                type="radio"
                value="light"
              />
              Light
            </label>
            <label
              class="check-label"
            >
              <input
                name="theme"
                type="radio"
                value="dark"
              />
              Dark
            </label>
          </form>
        </div>
      </header>
      <main
        class="main"
      >
        <div
          class="main__wrapper container"
        >
          <section
            class="content"
          >
            <h2>
              Loading...
            </h2>
          </section>
        </div>
      </main>
      <div
        class="flyout light-theme "
      >
        <div
          class="container flyout__wrapper"
        >
          <p
            class="flyout__text"
          >
            0 people are selected
          </p>
          <button
            class="button  flyout__btn"
            type="button"
          >
            Download
          </button>
          <button
            class="button button_error flyout__btn"
            type="button"
          >
            Unselect all
          </button>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
